woocommerce/woocommerce
# Product gallery: Enable block in blockified template and add transform from legacy one

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





As apart of this effort https://github.com/woocommerce/woocommerce/issues/42043 to surface the new Product Gallery block to users.

This PR transforms the Classic Single Product template to the blocks version with the new Product Gallery block. In addition, it adds the transform function for the legacy Product Image Gallery to the new Product Gallery Block.

Both of these additions will help surface this new Product Gallery block to existing and new users.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Test 1

1. Go to Appearance > Editor > Templates > Single Product (clear customizations)
2. Click on the editor and you should see "Transform into blocks" action button.
3. Hover over this button and you should see a preview with the Product Gallery block.
4. Click on the "Transform into blocks" action button and ensure you see the new Product Gallery block.

### Test 2

1. Go to Appearance > Editor > Templates > Single Product (clear customizations)
2. Click on the editor and you should see "Transform into blocks" action button.
3. Remove the new Product Gallery block.
4. Add the legacy Product Image Gallery block.
5. Click on the "transform to" button and ensure you see the Product Gallery block. ![file.png](https://github.com/woocommerce/woocommerce/assets/2132595/79a43780-56e6-4eda-870c-f5c17d99b24c)
6. Click on it and ensure your legacy Product Image Gallery block is now transformed to the new Product Gallery block.


### Test 3

1. Run `pnpm run --filter='@woocommerce/block-library' build:deploy` and then `pnpm run --filter='@woocommerce/plugin-woocommerce' build`.
2. Go to Appearance > Editor > Templates > Single Product (clear customizations)
3. Click on the editor and you should see "Transform into blocks" action button.
4. Ensure you only see the legacy Product Image Gallery and you can't transform to the new Product Gallery block.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
